coordinape/coordinape
# Migrate Daily Update Notifications to Hasura

closes #523 

This cron job pulls all the data it needs in one GQL query then sends
messages to all circles with an active epoch and a configured
notification channel.

I intentionally wrote this monolithically since it's a pretty unique
query and marshalling sequence as far as our codebase goes. If folks think
any of this code should be split out, I'd be happy to do so.

test plan:

- check if any circles have an active epoch. This sql should help
    ```sql
    select * from epoches
    where start_date < 'now()'
    and end_date > 'now()';
    ```

- if no epochs are active, adjust the dates of 1 or more existing epochs
  (for different circles) or try your luck by re-seeding the DB
- configure your test discord and/or telegram channel for the relevant
  circle(s)
- adjust the crontab config so the task runs minutely (* * * * *)
- collect a message and verify that the GIVE tokens, allocations totals
  and users who gifted in the last day match the database entries and
  totals.
- if no users have created pending gift entries in the last 24 hours you
  should touch the `updated` at field on some and verify the sender's
  name shows up ONCE in the new allocations today list.

If this is too general of a test plan, I'm happy to create a good static seed
for testing this and other things. I think we're reaching the point
where it makes sense to do this, and we'll be able to write integration
tests at that point as well.
